site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+const staticDirectory = './static';
+app.use('/static', express.static(staticDirectory));
+```
+
+We now have an item in the post dropdown menu that says "Create ticket", along with the icon located at `static/create-ticket.png` displayed next to the label. When the post menu item is clicked, our App will receive a [Call]({{< ref "call" >}}) at the endpoint `/forms/create-ticket-from-post/submit`. If we return a form in our response, a modal will be shown to the user with our form in the modal.
+
+Note that the `expand` object with `post: 'summary'` means that we will receive the relevant `Post` at `call.context.post` when we receive the request. This makes it so we don't have to fetch the `Post` manually when handling the request. Learn more about Expand [here]({{< ref "call" >}}).
+
+By defining the post menu item's `Call` with the path `/forms/create-ticket-post-menu`, we've implicitly registered an endpoint to process a request when this item is clicked:
+- `/forms/create-ticket-post-menu/submit`
+
+### Opening a modal
+
+Now let's process the post menu item click, and return an `AppForm` to show as a modal:
+
+```ts
+app.post('/forms/create-ticket-post-menu/submit', (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Extract the Post from the expanded context
+    const post: Post = call.context.post;
+
+    const call: AppCall = {
+        path: '/forms/create-ticket',
+        expand: {
+            acting_user: 'summary', // Expand the acting user so we can get the user's username for our post
+            team: 'summary', // Expand the current team so we can construct a permalink for the selected post using the team's name
+        },
+    };
+
+    const form: AppForm = {
+        title: 'Create a Ticket',
+        fields: [
+            {
+                name: 'summary',
+                value: '',
+                modal_label: 'Summary',
+                type: 'text',
+                is_required: true, // This makes it so the user must provide this value before submitting the form
+                min_length: 10, // We set a minimum length of 10 characters for this field
+            },
+            {
+                name: 'description',
+                value: post.message, // Use the Post's message as the default value for the description field
+                modal_label: 'Description',
+                type: 'text',
+                subtype: 'textarea',
+            },
+        ],
+    };
+
+    const callResponse: AppCallResponse = {
+        type: 'form',
+        call,
+        form,
+    }
+
+    res.json(callResponse);
+});
+```
+
+The user is shown a form with text fields labeled "Summary" and "Description". The Description field is prepopulated with the `Post`'s message.
+
+By defining the form's `Call` with the path `/forms/create-ticket`, we've implicitly registered three endpoints for this form to use:
```

Comment:
might help to specify/mention if these are the only/always three endpoints of if there are more (or less?, depending) for other forms.